### ISSUE TITLE
[NV-1525] 🚀 Feature: retry option for failed step in workflow

### ISSUE THREAD

**Issue by jainpawan21** [novuhq/novu#2405]
### 🔖 Feature description

Theres should be some retry policy for failed step in workflow

### 🎤 Why is this feature needed ?

There can be possibility of service down, so retry policy will ensure that step is executed in finite number of retries.

### ✌️ How do you aim to achieve this?

NA

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

<sub>[NV-1525](https://linear.app/novu/issue/NV-1525/🚀-feature-retry-option-for-failed-step-in-workflow)</sub>

**Comment by szbartnik** [comment]
@scopsy Is it really not planned? We've just observed that we had failed step in our workflow:
![image](https://github.com/novuhq/novu/assets/5238147/eff1f9b2-120f-46cb-b2c7-17d4835c15a3)

Errors can always happen so sounds like it's worth to have a way to retry failed steps in a failed workflow. 

Btw. when error happens is there any way to get notified about it? Because otherwise it can be easily missed?
